components/Alert/AlertProvider.jsx

```diff
@@ -0,0 +1,49 @@
+import { createContext, useState, useRef, useEffect } from "react";
+
+const initialState = {
+    text: "",
+    color: "",
+    icon: "",
+    totalAlerts: 0,
+};
+
+export const AlertContext = createContext({ ...initialState, setAlert: () => {} });
+
+export const AlertProvider = ({ children }) => {
+    const [text, setText] = useState(initialState.text);
+    const [color, setColor] = useState(initialState.color);
+    const [icon, setIcon] = useState(initialState.icon);
+    const [alertNo, setAlertNo] = useState(initialState.totalAlerts);
+    const alertRef = useRef(alertNo);
```

Comment:
alertRef and alertNo is definitely confusing me. My best guess: is it because setTimeout needs to be independent of renders?

Reply:
Explained in person